tket2-hseries/src/llvm/array_utils.rs

```diff
@@ -3,13 +3,66 @@
 // TODO move to hugr-llvm crate
 // https://github.com/CQCL/tket2/issues/899
 use anyhow::Result;
-use hugr::llvm::inkwell;
+use hugr::llvm::extension::collections::array::decompose_array_fat_pointer;
+use hugr::llvm::extension::collections::{array, stack_array};
+use hugr::llvm::inkwell::values::BasicValueEnum;
+use hugr::llvm::{inkwell, CodegenExtsBuilder};
+use hugr::{HugrView, Node};
 use inkwell::builder::{Builder, BuilderError};
 use inkwell::context::Context;
 use inkwell::types::{IntType, PointerType, StructType};
 use inkwell::values::{ArrayValue, IntValue, PointerValue, StructValue};
 use inkwell::AddressSpace;
 
+/// Different array lowering strategies.
+#[derive(Clone, Copy, Debug, Default)]
+pub enum ArrayLowering {
```

Comment:
This would arguably fit better into the Hugr repo, but then we would need a separate Hugr release to make this available...

Reply:
we can do a patch hugr release fairly shortly if it is a non-breaking addition